bip-0066.mediawiki

```diff
@@ -55,24 +55,30 @@ bool static IsValidSignatureEncoding(const std::vector<unsigned char> &sig) {
     // Make sure the length covers the entire signature.
     if (sig[1] != sig.size() - 3) return false;
 
+    // Check whether the R element is an integer.
+    if (sig[2] != 0x02) return false;
+
     // Extract the length of the R element.
     unsigned int lenR = sig[3];
 
-    // Make sure the length of the S element is still inside the signature.
+    // Zero-length integers are not allowed for R.
+    if (lenR == 0) return false;
+
+    // Make sure the length of the R element is still inside the signature.
```

Comment:
We are checking that `0x30 [total-length] 0x02 [R-length] [R] 0x02` is still inside the signature, `5 + lenR`.
I therefore changed this to `Make sure the length of the R element is still inside the signature`